### DISCUSSION TITLE
Duplicated function declaration error with @babel/plugin-syntax-top-level-await

### DISCUSSION THREAD

**Discussion by liuliangsir** [babel/babel#12434] [Q&A] [answered]
**Current behavior**
- [Babel REPL](https://babeljs.io/repl#?browsers=ie%2010&build=&builtIns=false&spec=false&loose=false&code_lz=IYd2EsBcAJgYwHRwE4FNiVQeQA6XAPYB2A4sgQK44AUA3gM4BKq9AkgCYBc0A5ACQAOAEwB2ACw8AvgEoA3ADMKROPmLR5BAtGrTotSeqUrCRdZu279QA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=unambiguous&lineWrap=true&presets=env&prettier=false&targets=&version=7.12.9&externalPlugins=%40babel%2Fplugin-syntax-top-level-await%407.12.1)

**Comment by liuliangsir** [comment]
@nicolo-ribaudo @JLHwung cc

**Comment by nicolo-ribaudo** [comment, marked as answer]
It's because in ES modules you cannot have duplicate function declarations.
Top-level await is only allowed in modules, so `"sourceType": "unambiguous"` is interpreted as `"sourceType": "module"` in your example.

**Reply by liuliangsir** [reply]
@nicolo-ribaudo is there any workaround to resolve the issue above? Remove `@babel/plugin-syntax-top-level-await` from babel configuration file?

**Reply by nicolo-ribaudo** [reply]
You can either avoid using top-level await in that file, or two functions with the same name: they are incompatible features, similarly to `with (...) {}` and `"use strict"`.

**Reply by liuliangsir** [reply]
Okay, thanks!
